src/providers/seventv/eventapi/SeventvEventAPIClient.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "providers/liveupdates/BasicPubSubClient.hpp"
+#include "providers/seventv/eventapi/SeventvEventAPISubscription.hpp"
+
+namespace chatterino {
+
+class SeventvEventAPIClient
+    : public BasicPubSubClient<SeventvEventAPISubscription>
+{
+public:
+    SeventvEventAPIClient(liveupdates::WebsocketClient &websocketClient,
+                          liveupdates::WebsocketHandle handle,
+                          std::chrono::milliseconds heartbeatInterval);
+
+    void setHeartbeatInterval(int intervalMs);
+    void handleHeartbeat();
+
+protected:
+    void onConnectionEstablished() override;
+
+private:
+    void checkHeartbeat();
+
+    std::atomic<std::chrono::time_point<std::chrono::steady_clock>>
+        lastHeartbeat_;
+    std::atomic<std::chrono::milliseconds> heartbeatInterval_;
```

Comment:
See if we can get away with this not being an atomic as it's only going to be set by the constructor, by the initial handshake, and then only read